packages/cart/src/services/cart-module.ts

```diff
@@ -143,4 +148,57 @@ export default class CartModuleService implements ICartModuleService {
   ): Promise<void> {
     await this.cartService_.delete(ids, sharedContext)
   }
+
+  @InjectTransactionManager("baseRepository_")
+  async listAddresses(
+    filters: FilterableAddressProps = {},
+    config: FindConfig<AddressDTO> = {},
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const addresses = await this.addressService_.list(
+      filters,
+      config,
+      sharedContext
+    )
+
+    return await this.baseRepository_.serialize<AddressDTO[]>(addresses, {
+      populate: true,
+    })
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async createAddresses(
+    data: CreateAddressDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const addresses = await this.addressService_.create(data, sharedContext)
+
+    return await this.listAddresses(
+      { id: addresses.map((p) => p!.id) },
+      {},
+      sharedContext
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async updateAddresses(
+    data: UpdateAddressDTO[],
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    const addresses = await this.addressService_.update(data, sharedContext)
+
+    return await this.listAddresses(
+      { id: addresses.map((p) => p!.id) },
+      {},
+      sharedContext
+    )
+  }
+
+  @InjectTransactionManager("baseRepository_")
+  async deleteAddresses(
+    ids: string[],
+    @MedusaContext() sharedContext: Context = {}
+  ) {
+    await this.addressService_.delete(ids, sharedContext)
+  }
```

Comment:
comment: will we need soft deletion and restoration?

Reply:
I am not sure yet. Will address this in a follow up PR, if needed.